mantidproject/mantid
# Prevent duplicate spectra in Indirect F(Q)Fit Multiple Table

**Description of work.**
Code altered to prevent spectra duplicates in the F(Q)Fit Multiple Spectra Table. Removed obsolete methods and tests.

**To test:**

Test 1
1) Open Indirect Data Analysis
2) go to the F(Q) fit tab and switch to multiple input mode.
3) click add workspace, browse to a suitable file and open
4) Add different Width and EISF parameters to the table. Try adding ones that have already been added - should not duplicate them.

Test 2
1) Open Indirect Data Analysis
2) go to the F(Q) fit tab and stay on the single input tab
3) click on browse, find a suitable file and open
4) Check that the Fit Parameter and Width/EISF drop downs work.

Fixes #29244  

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
